File: solutions/beecrowd/2203/2203.py
import sys
from math import sqrt

for line in sys.stdin:
    xf, yf, xi, yi, vi, r1, r2 = map(int, line.split())
    distance = sqrt(pow(xi - xf, 2) + pow(yi - yf, 2)) + 1.5 * vi
    if distance <= r1 + r2:
        print('Y')
    else:
        print('N')

File: solutions/beecrowd/2203/2203.clj
(ns main
  (:require [clojure.string :as str]))

(defn main []
  (loop [line (read-line)]
    (when line
      (let [[xf yf xi yi vi r1 r2] (map #(Integer/parseInt %) (str/split line #" "))
            distance (-> (- xi xf)
                         (Math/pow 2)
                         (+ (Math/pow (- yi yf) 2))
                         (Math/sqrt)
                         (+ (* vi 1.5)))]
        (if (<= distance (+ r1 r2))
          (println "Y")
          (println "N")))
      (recur (read-line)))))

(main)
